packages/block-editor/src/components/use-block-commands/index.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	hasBlockSupport,
+	store as blocksStore,
+	switchToBlockType,
+	isTemplatePart,
+} from '@wordpress/blocks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useCommandLoader } from '@wordpress/commands';
+import {
+	copy,
+	edit as remove,
+	create as add,
+	group,
+	ungroup,
+	moveTo as move,
+} from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+export const useTransformCommands = () => {
+	const { clientIds } = useSelect( ( select ) => {
+		const { getSelectedBlockClientIds } = select( blockEditorStore );
+		const selectedBlockClientIds = getSelectedBlockClientIds();
+
+		return {
+			clientIds: selectedBlockClientIds,
+		};
+	}, [] );
+	const blocks = useSelect(
+		( select ) =>
+			select( blockEditorStore ).getBlocksByClientId( clientIds ),
+		[ clientIds ]
+	);
+	const { replaceBlocks, multiSelect } = useDispatch( blockEditorStore );
+	const { possibleBlockTransformations, canRemove } = useSelect(
+		( select ) => {
+			const {
+				getBlockRootClientId,
+				getBlockTransformItems,
+				canRemoveBlocks,
+			} = select( blockEditorStore );
+			const rootClientId = getBlockRootClientId(
+				Array.isArray( clientIds ) ? clientIds[ 0 ] : clientIds
+			);
+			return {
+				possibleBlockTransformations: getBlockTransformItems(
```

Comment:
In some contexts, this is triggering failures in the command loader. I'm guessing when the "blocks" array is empty.

cc @draganescu 

Also @artpi for more details.

Reply:
I'm still tracking down the exact reproduction path, but here is a stack trace I am getting when I run command loader quite often (for a different env that command pallette):

![image](https://github.com/WordPress/gutenberg/assets/3775068/af07eb71-b095-4a0b-a7a9-7126d40717c0)

Reply:
I am not sure what to do to trigger a visible error. Ultimately empty `blocks` is guarded in `getPossibleBlockTransformations` by:

```
if ( ! blocks.length ) {
		return [];
	}
```

Reply:
Yes, I see that in the code and yet I get the above stack trace. My suspicion is that we're getting `[null, null ]` or something similar

Reply:
I finally have a reproduciton path. It took me quite a while:

https://github.com/WordPress/gutenberg/issues/59289